packages/framework/src/Component/Plugin/PluginCrudExtensionFacade.php

```diff
@@ -28,22 +28,23 @@ public function __construct(PluginCrudExtensionRegistry $pluginCrudExtensionRegi
      */
     public function extendForm(FormBuilderInterface $builder, $type, $name)
     {
-        $builder->add($name, FormType::class, [
-            'compound' => true,
-            'render_form_row' => false,
-        ]);
-
         $crudExtensions = $this->pluginCrudExtensionRegistry->getCrudExtensions($type);
+
         foreach ($crudExtensions as $key => $crudExtension) {
             $builderExtensionGroup = $builder->create($key . 'Group', GroupType::class, [
                 'label' => $crudExtension->getFormLabel(),
             ]);
 
-            $builderExtensionGroup->add($key, $crudExtension->getFormTypeClass(), [
+            $builderExtensionGroup->add($name, FormType::class, [
+                'compound' => true,
+                'render_form_row' => false,
+            ]);
```

Comment:
The compound `FormType` has no other meaning here but to include the `$name` in the property path so it is mapped eg. to `CategoryData::$pluginData['shopsysProductFeedHeurekaBundleFormCategoryCrudExtension']`. You can achieve this much easily without the extra form using [the `property_path` option](https://symfony.com/doc/current/reference/forms/types/form.html#property-path) like this:

```php
$builderExtensionGroup->add($key, $crudExtension->getFormTypeClass(), [
    'render_form_row' => false,
    'property_path' => sprintf('%s[%s]', $name, $key),
]);
```

*(this code would replace the two form fields defined in lines 38-45, it behaved fine during testing)*

I can create a fixup commit for this if you want to - I didn't want to push it before I discussed it with you.

Reply:
@PetrHeinz fixup would be nice, thanks.

Reply:
@PetrHeinz thank you for your fixup, I like it :)